### ISSUE TITLE
Reviewing agent log page shows spinning gif and "Just a moment..." forever; logs show "500 Internal Server Error"

### ISSUE THREAD

**Issue by ohingardail** [cantino/huginn#1633]
When I try to review the log page of some (but not all) of my de-duplication agents, the page shows the spinning gif and a "Just a moment..." text forever. A log shows:
```
==> production.log <==
I, [2016-08-15T09:54:17.599996 #14537]  INFO -- : [25f05abb-1fd8-4e6c-852f-97d6b1cde9c1] Started GET "/agents/167/logs" for fe80::f6:960d:48a:d26 at 2016-08-15 09:54:17 +0100
I, [2016-08-15T09:54:17.601867 #14537]  INFO -- : [25f05abb-1fd8-4e6c-852f-97d6b1cde9c1] Processing by LogsController#index as */*
I, [2016-08-15T09:54:17.601936 #14537]  INFO -- : [25f05abb-1fd8-4e6c-852f-97d6b1cde9c1]   Parameters: {"agent_id"=>"167"}
I, [2016-08-15T09:54:17.799857 #14537]  INFO -- : [25f05abb-1fd8-4e6c-852f-97d6b1cde9c1]   Rendered logs/index.html.erb (192.4ms)
I, [2016-08-15T09:54:17.800055 #14537]  INFO -- : [25f05abb-1fd8-4e6c-852f-97d6b1cde9c1] Completed 500 Internal Server Error in 198ms (ActiveRecord: 137.4ms)
F, [2016-08-15T09:54:17.800987 #14537] FATAL -- : [25f05abb-1fd8-4e6c-852f-97d6b1cde9c1] 
ActionView::Template::Error (No route matches {:action=>"show", :controller=>"events", :id=>nil} missing required keys: [:id]):
    14:         <td>
    15:           <div class="btn-group btn-group-xs">
    16:             <% if log.inbound_event_id.present? %>
    17:               <%= link_to 'Event In', event_path(log.inbound_event), class: "btn btn-default" %>
    18:             <% else %>
    19:               <%= link_to 'Event In', '#', class: "btn btn-default disabled" %>
    20:             <% end %>
  app/views/logs/index.html.erb:17:in `block in _app_views_logs_index_html_erb___4052024855779755188_47401770126020'
  app/views/logs/index.html.erb:9:in `_app_views_logs_index_html_erb___4052024855779755188_47401770126020'
  app/controllers/logs_controller.rb:6:in `index'
  config/initializers/silence_worker_status_logger.rb:5:in `call_with_silence_worker_status'
```

**Comment by cantino** [comment]
Thanks @ohingardail, this looks like a bug.

**Comment by cantino** [comment]
Would you mind seeing if #1634 fixes it for you @ohingardail?

**Comment by ohingardail** [comment]
The fix in #1634 seems to work; I've poked around in a random selection of other agent type's log pages and they seem to work too.

**Comment by cantino** [comment]
Thanks for testing it!
